packages/terra-data-grid/src/DataGrid.jsx

```diff
@@ -195,6 +196,25 @@ const DataGrid = forwardRef((props, ref) => {
   } = props;
 
   const displayedColumns = (hasSelectableRows ? [WorklistDataGridUtils.ROW_SELECTION_COLUMN] : []).concat(pinnedColumns).concat(overflowColumns);
+  // Create new displayedColumns object to account for column spans
+  const displayedColumnsWithColumnSpan = useMemo(() => {
+    let i = 0;
+    const newDisplayedColumns = [];
+    displayedColumns.forEach((column) => {
```

Comment:
The forEach function provides an index as a parameter to its callback function.  Is there a reason for the "i" variable that is not covered by this available argument?

Reply:
The length of the new columns object we are creating is not equal to the original length of columns because of spans. So we do need the `i` variable.